yarn-project/p2p/src/client/p2p_client.ts

```diff
@@ -202,6 +220,22 @@ export class P2PClient implements P2P {
     this.synchedProvenBlockNumber = store.openSingleton('p2p_pool_last_proven_l2_block');
   }
 
+  #assertIsReady() {
+    if (!this.isReady()) {
+      throw new Error('P2P client not ready');
+    }
+  }
+
+  getEpochProofQuotes(epoch: bigint): Promise<EpochProofQuote[]> {
```

Comment:
It feels peculiar that we query the p2p client for quotes. Maybe this is how the tx pool was implemented and the pattern has stuck. But wouldn't it make more sense for the pools to be independent?

Reply:
I think the main reason it smells here is because we know we're going to need to backchannel quotes into it from http.

That aside, the p2p client is the only way that things get into the pools, so I think its okay that they are owned by it.

And since we know that the backchannel is going to be removed (or at least unused in production), I'm reluctant to refactor them out.

Reply:
i do think we should plan at some point in the future to have the pools be owned externally, then have them passed into the p2p client.